### PR TITLE
added descriptions of CLI args for configuring Execution Engine options

### LINKED ISSUE #966: Document config options for Execution Engine functionality
## Describe the bug

There are many newly introduced configuration options for setting up Besu as an Execution Engine for PoS Ethereum on mainnet.

- [X] Missing content
- [ ] Outdated content
- [ ] Wrong content
- [ ] Confusing or misleading content
- [ ] Other

## Change suggestion

New functionality needs to be documented.

### PR DESCRIPTION
**Pull request by jflo**
Signed-off-by: Justin Florentine <justin+github@florentine.us>

## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [X] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [X] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [X] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [ ] you fixed all the issues raised by the tests, if any.
- [ ] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change
Describes new config options for setting up Besu as an Execution Engine.

## Issue fixed
fixes #966

### For content changes

- [X] Doc content
- [ ] Doc pages organisation

## Testing

https://hyperledger-besu--966.org.readthedocs.build/en/966/

### CODE REVIEW

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -394,6 +394,151 @@ The default is `true`.
...
+    BESU_ENGINE_RPC_HTTP_PORT=8550
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    engine-rpc-http-port=8550
+    ```
+
+The listening port for the Engine API calls (ENGINE, ETH) for JSON-RPC over HTTP.
```
Comment (rolandtyler) [inline docs/Reference/CLI/CLI-Syntax.md:423]: Is there a default port?

File: docs/Reference/CLI/CLI-Syntax.md [hunk 2]
```diff
@@ -394,6 +394,151 @@ The default is `true`.
...
+    BESU_ENGINE_RPC_WS_PORT=8551
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    engine-rpc-ws-port=8551
+    ```
+
+The listening port for the Engine API calls (ENGINE, ETH) for JSON-RPC over WebSockets.
```
Comment (rolandtyler) [inline docs/Reference/CLI/CLI-Syntax.md:451]: Same, is there a default port?

**Review by rolandtyler — COMMENTED** [review]
@jflo, Pushed a few edits for formatting consistency, and left a few questions.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:423]:
```suggestion
The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over HTTP.
```

File: docs/Reference/CLI/CLI-Syntax.md (hunk 2 above)
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:451]:
```suggestion
The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over WebSockets.
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 3]
```diff
@@ -394,6 +394,151 @@ The default is `true`.
...
+    ```bash
+    BESU_ENGINE_JWT_ENABLED=true
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    engine-jwt-enabled=true
+    ```
+Enables or disables authentication for Engine APIs. Defaults to `false`.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:511]:
```suggestion
Enables or disables authentication for Engine APIs. The default is `false`.
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 4]
```diff
@@ -394,6 +394,151 @@ The default is `true`.
...
+    BESU_ENGINE_HOST_ALLOWLIST=localhost,127.0.0.1
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    engine-host-allowlist=localhost,127.0.0.1
+    ```
+
+A comma separated list of hostnames to allow for ENGINE API access (applies to both HTTP and WebSockets).
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:479]:
```suggestion
A comma-separated list of hostnames to allow for `ENGINE` API access (applies to both HTTP and WebSockets).
```

**Review by alexandratran — APPROVED** [review]
Push some edits. LGTM.

**Review by macfarla — COMMENTED** [review]
CLI additions look good. idk if the package-lock.json file is meant to be included in this change?

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (macfarla) [reply docs/Reference/CLI/CLI-Syntax.md]: default is 8550

File: docs/Reference/CLI/CLI-Syntax.md (hunk 2 above)
Comment (macfarla) [reply docs/Reference/CLI/CLI-Syntax.md]: 8551

File: docs/Reference/CLI/CLI-Syntax.md [hunk 5]
```diff
@@ -394,6 +394,151 @@ The default is `true`.
...
+    BESU_ENGINE_RPC_HTTP_PORT=8550
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    engine-rpc-http-port=8550
+    ```
+
+The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over HTTP.
```
Comment (bgravenorst) [inline docs/Reference/CLI/CLI-Syntax.md:423]: @jflo, Are Engine API calls, calls made from a consensus client (Teku)? Are these a seperate set of APIs that are implemented for the merge?

**Pull request merged**